oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticIndexWriter.java

```diff
@@ -254,15 +261,35 @@ public void afterBulk(long executionId, BulkRequest bulkRequest, BulkResponse bu
                 }
             }
             if (bulkResponse.hasFailures()) { // check if some operations failed to execute
+
+                NodeBuilder status = definitionBuilder.child(IndexDefinition.STATUS_NODE);
+                // Read the current failed paths (if any) on the :status node into failedDocList
+                if (status.hasProperty(IndexDefinition.FAILED_DOC_PATHS)) {
+                    for (String str : status.getProperty(IndexDefinition.FAILED_DOC_PATHS).getValue(Type.STRINGS)) {
+                        failedDocSet.add(str);
+                    }
+                }
+
                 for (BulkItemResponse bulkItemResponse : bulkResponse) {
                     if (bulkItemResponse.isFailed()) {
                         BulkItemResponse.Failure failure = bulkItemResponse.getFailure();
-                        LOG.error("Bulk item with id {} failed", failure.getId(), failure.getCause());
+                        failedDocSet.add(bulkItemResponse.getId());
+                        // Log entry to be used to parse logs to get the failed doc id/path if needed
+                        LOG.error("ElasticIndex Update Doc Failure: Error while adding/updating doc with id : [{}]", bulkItemResponse.getId());
+                        LOG.error("Failure Details: BulkItem ID: " + failure.getId() + ", Failure Cause: {}", failure.getCause());
                     } else {
                         // Set indexUpdated to true even if 1 item was updated successfully
                         updatesMap.put(executionId, Boolean.TRUE);
                     }
                 }
+
+                if (failedDocSet.size() == FAILED_DOC_COUNT_FOR_STATUS_NODE) {
+                    LOG.info("Failed Docs count exceeds the persistence limit. Will skip persisting paths of more failed docs." +
+                            "Please analyze logs to identify the failing docs. Search for ElasticIndex Update Doc Failure");
```

Comment:
It seems a bit weird to me to state in the log to have a look in the same log that you are looking at. Maybe something like:
`Failing docs should be mentioned above under ElasticIndex Update Doc Failure`
instead of 
`Please analyze logs to identify the failing docs. Search for ElasticIndex Update Doc Failure`